server/repository/src/db_diesel/number_row.rs

```diff
@@ -40,6 +40,38 @@ impl fmt::Display for NumberRowType {
             NumberRowType::RequestRequisition => write!(f, "REQUEST_REQUISITION"),
             NumberRowType::ResponseRequisition => write!(f, "RESPONSE_REQUISITION"),
             NumberRowType::Stocktake => write!(f, "STOCKTAKE"),
+            NumberRowType::Program(custom_string) => write!(f, "PROGRAM_{}", custom_string),
+        }
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum NumberRowTypeError {
+    UnknownTypePrefix(String),
+    MissingTypePrefix,
+}
+
+impl TryFrom<String> for NumberRowType {
+    type Error = NumberRowTypeError;
+
+    fn try_from(s: String) -> Result<Self, NumberRowTypeError> {
+        match s.as_str() {
+            "INBOUND_SHIPMENT" => Ok(NumberRowType::InboundShipment),
+            "OUTBOUND_SHIPMENT" => Ok(NumberRowType::OutboundShipment),
+            "INVENTORY_ADJUSTMENT" => Ok(NumberRowType::InventoryAdjustment),
+            "REQUEST_REQUISITION" => Ok(NumberRowType::RequestRequisition),
+            "RESPONSE_REQUISITION" => Ok(NumberRowType::ResponseRequisition),
+            "STOCKTAKE" => Ok(NumberRowType::Stocktake),
+            _ => match s.rsplit_once('_') {
```

Comment:
this should be split_once not rsplit_once, right?

Reply:
Or use a different delimiter, e.g.:

`NOT_A_PROGRAM#number_name`

Reply:
Yes you are right good catch! For some reason I thought it was the reverse.